pkg/kubelet/status/status_manager.go

```diff
@@ -178,7 +178,7 @@ func (m *manager) GetPodStatus(uid types.UID) (v1.PodStatus, bool) {
 	return status.status, ok
 }
 
-func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
+func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) bool {
```

Comment:
We don't have to address this issue in this diff, but I think that it feels a little weird to me to have the `SetPodStatus` and `updateStatusInternal` methods return `bool` to indicate success/failure, instead of returning `error`.

But, I'm not sure this relates directly to your change to check container state transitions for ephemeral containers, so I'm fine leaving it for now.

Reply:
Returning bool suffices for allowing unit test to assert the fix.
If, in the future, we need to distinguish various scenarios, we can change the return type to error.

Reply:
Ok - I could see arguments both for and against returning the `bool` strictly for testing purposes. I still think we could assert the conditions necessary for the test using only the `verifyUpdates` method; that's what the tests did prior to this diff.

But at the end of the day, I don't feel super strongly.